docs/sdks/server-sdks/ruby.md

```diff
@@ -84,71 +138,152 @@ The SDK will invoke the `log_assignment` function with an `assignment` object th
 
 ```ruby
 require 'eppo_client'
-require 'logger'
 
 client = EppoClient::Client.instance
-variation = client.get_string_assignment(
-  '<SUBJECT-KEY>',
-  '<FLAG-KEY>',
-  {},
-  Logger::DEBUG
+bandit_result = client.get_bandit_action(
+  "shoe-bandit",
+  name,
+  EppoClient::Attributes.new(
+    numeric_attributes: { "age" => age }, categorical_attributes: { "country" => country }
+  ),
+  {
+    "nike" => EppoClient::Attributes.new(
+      numeric_attributes: { "brand_affinity" => 2.3 },
+      categorical_attributes: { "image_aspect_ratio" => "16:9" }
+    ),
+    "adidas" => EppoClient::Attributes.new(
+      numeric_attributes: { "brand_affinity" => 0.2 },
+      categorical_attributes: { "image_aspect_ratio" => "16:9" }
+    )
+  },
+  "control"
 )
 ```
 
-<br />
+#### Subject Context
+
+The subject context contains contextual information about the subject that is independent of bandit actions.
+For example, the subject's age or country.
+
+The subject context has type `Attributes` which has two fields:
+
+- `numeric_attributes` (Hash{String => Float}): A hash of numeric attributes (such as "age")
+- `categorical_attributes` (Hash{String => String}): A hash of categorical attributes (such as "country")
 
 :::note
-It may take up to 10 seconds for changes to Eppo experiments to be reflected by the SDK assignments.
+The `categerical_attributes` are also used for targeting rules for the feature flag similar to how `subject_attributes` are used for that with regular feature flags. 
 :::
+
+#### Action Contexts
+
+Next, supply a hash with actions and their attributes: `actions: Hash{String => Attributes}`.
+If the user is assigned to the bandit, the bandit selects one of the actions supplied here,
+and all actions supplied are considered to be valid; if an action should not be shown to a user, do not include it in this hash.
+
+The action attributes are similar to the `subject_attributes` but hold action-specific information.
+Note that we can use `Attributes.empty` to create an empty attribute context.
+
+Note that action contexts can contain two kinds of information:
+- Action-specific context: e.g., the image aspect ratio of the image corresponding to this action
+- User-action interaction context: e.g., there could be a "brand-affinity" model that computes brand affinities of users to brands, and scores of this model can be added to the action context to provide additional context for the bandit.
+
+#### Result
+
+The `bandit_result` is an instance of `BanditResult`, which has two fields:
+
+- `variation` (String): The variation that was assigned to the subject
+- `action` (Optional[String]): The action that was assigned to the subject
+
+The variation returns the feature flag variation; this can be the bandit itself, or the "status quo" variation if the user is not assigned to the bandit.
+If we are unable to generate a variation, for example when the flag is turned off, then the `default` variation is returned. 
```

Comment:
There is another case, where the bandit encounters an error selecting an action (or there are no actions provided) in which case it will return the bandit variation, but a `nil` action. I wonder if that's worth explicitly calling out--I think it is.